src/style/fog.js

```diff
@@ -53,16 +60,25 @@ export class FogSampler {
         // function is C2 continuous at the onset. The fog is about 99% opaque at
         // the far limit, so we simply scale it and clip to achieve 100% opacity.
         // https://www.desmos.com/calculator/3taufutxid
-        // The output of this function must match src/shaders/_prelude_fog.fragment.glsl
-        const decay = 5.5;
-        let falloff = Math.max(0.0, 1.0 - Math.exp(-decay * (depth - start) / (end - start)));
+        // The output of this function should match src/shaders/_prelude_fog.fragment.glsl
+        const decay = 6;
+        const t = (depth - start) / (end - start);
+        let falloff = 1.0 - Math.min(1, Math.exp(-decay * t));
 
         // Cube without pow()
         falloff *= falloff * falloff;
 
         // Scale and clip to 1 at the far limit
         falloff = Math.min(1.0, 1.00747 * falloff);
 
+        // From src/render/painter.js via fog uniforms:
+        const fogExponent = 12 * Math.pow(1 - fogStrength, 2);
+
+        // Account for fog strength
+        falloff *= Math.pow(smoothstep(0, 1, t), fogExponent);
+
+        // We may wish to account for haze's effect on obscuring symbols
```

Comment:
This comment may be extraneous since this function can be used for other purposes than symbols and exposed in the public API, it may be misleading.

```suggestion
```

Reply:
👍  I initially had `TODO:` but the linter rejected it. I debated adding `getHazeOpacity` separately. I suspect the full answer is somewhere between the two since I still think symbol opacity precisely following fog opacity is an arbitrary choice we may not actually want once we assemble more pieces.